docs/concepts/mixer.md

```diff
@@ -7,40 +7,210 @@ title: Mixer
+          value_type: 1 # STRING
+        - name: method
+          value_type: 1 # STRING
+        - name: response_code
+          value_type: 2 # INT64
+
+This is declaring a descriptor called request_count. Because of the kind and value fields, policy objects associated with this
+descriptor represent 64-bit integer counters. Additionally, each policy object is uniquely identified via the 5 listed labels.
+Producing a policy object for such a descriptor requires 6 pieces of information:
+
+- a 64-bit integer metric value 
+- a source string
+- a target string
+- a service string
+- a method string
+- a 64-bit integer response code
+
+Here is an example snippet of Istio configuratin which produces a policy object for the above descriptor:
+
+      metrics:
+      - descriptor_name: request_count
+        value: "1"
+        labels:
+          source: source.name | "unknown"
+          target: target.name | "unknown"
+          service: api.name | "unknown"
+          method: api.method | "unknown"
+          response_code: response.code | 200
+
+Many such policy objects are created as part of attribute processing and they ultimately serve as input to
+adapters.
+
+Explicitly defining descriptors and creating policy objects for them is akin to types and objects in a traditional
+programming language. Doing so enables a few important scenarios:
+
+- Having the set of descriptors explicitly defined enables Istio to program backend systems to accept traffic produced
+by the mixer. For example, a MetricDescriptor provides all the information needed to program a backend system to accept metrics
+that conform to the descriptor's shape (it's value type and its set of labels).
+
+- It enables type checking of the deployment's configuration. Since attributes have strong types, and so do descriptors,
+Istio can provide a number of strong correctness guarantees of the system's configuration. Basically, if a chunk of
+configuration is accepted into the Istio system, it means the configuration passes a minimum correctness bar. Again, this
+plays the same role as types in a programming language.
+
+- It enables us to provide a strongly-typed scripting environment.
+
+## Scripting
+
+The mixer's attribute processing phase is implemented via a scripting language (exact language *TBD*). 
+The scripts are provided a set of attributes and are responsible for producing policy objects and dispatching
+control to individual configured aspects.
+
+For common uses, the operator authors policy objects production rules via a relatively simple declarative format
+and expression syntax. The mixer ingests such rules and produces a script that performs the necessary runtime work
+of accessing the request's incoming attributes and producing the requisite policy objects.
+
+For advanced uses, the operator can bypass the declarative format and author directly in the scripting
+language. This is more complex, but provides ultimate flexibility.
+
 ## Adapters
```

Comment:
I would also move this section above the "Request phases" bit.  Get all the concepts out of the way before talking about how they are woven together.